4 - AnimalShelter/Code/OperationResult.cs

```diff
@@ -0,0 +1,25 @@
+﻿using AnimalShelterCode;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AnimalShelterCode
+{
+    public class OperationResult
+    {
+        public OperationResult(Animal animal)
+        {
+            Animal = animal;
+            OperationSuccess = true;
+        }
+        public OperationResult(string error)
+        {
+            Error = error;
+            OperationSuccess = false;
+        }
+        internal Animal Animal { get; private set; }
+        internal string Error { get; private set; }
+        internal bool OperationSuccess { get; private set; }
```

Comment:
I find it interesting you are using "internal" here. Internal means "can only be accessed by the assembly this type is defined in". It should only be used when creating a public DLL where you don't want to expose "hooking functionality" to those using the shared code.